### ISSUE TITLE
OpenAPI: Bad reference

### ISSUE THREAD

**Issue by dbluhm** [hyperledger/aries-cloudagent-python#3086]
https://github.com/hyperledger/aries-cloudagent-python/blob/8d4ee585097f1f6f12427de4bdefdf8dda272678/open-api/openapi.json#L9739

The IndyCredential schema is referencing IndyAttrValue but the reference is to the wrong location; `#/definitions/IndyAttrValue` when it should be `#/components/schemas/IndyAttrValue`. It's not clear exactly why the Schemas are behaving this way; all other schema refs are correct.

@ff137 This might be related to changes in #2894? Any thoughts?

**Comment by ff137** [comment]
That's very strange ... the `#/definitions/IndyAttrValue` reference is how it is autogenerated with `generate-open-api-spec`. That seems to be the Swagger (OpenAPI 2.0) convention (all references in swagger.json are like that). In OpenAPI 3 it should indeed be `#/components/schemas/IndyAttrValue`, and openapi.json has that one as a `#/definitions` reference.

Even upon regenerating the spec, it looks the same. And none of the errors or warnings in the output logs seem to be related to that schema object. But there are a few errors reported cryptically, so can't be sure:
```
ERROR io.swagger.codegen.DefaultCodegen - No Type defined for Additional Property io.swagger.models.properties.BooleanValueProperty@4cf
        In Property: io.swagger.models.properties.MapProperty@403ca3f0
```

Not sure what's going on there. Can't see anything wrong in that PR, but it is the only schema object that extends `fields.Dict`, and I can't remember how I figured that one out ... I'll take a closer look and get back to you

**Comment by ff137** [comment]
As far as I can tell, it's a bug with the openapi generator. Because the Swagger 2.0 spec is correct, but it doesn't translate properly to OpenAPI 3.0

The same behavior occurs when upgrading the openapi generator to latest.

Ah, while testing, I managed to fix it! 

The problem is actually in the IndyCredential model:
it has a `values` field, which is a custom Dictionary, that we define to have types: keys=Str and values=IndyAttrValue.
This generates in Swagger 2.0:
```json
          "values" : {
            "additionalProperties" : {
              "allOf" : [ {
                "$ref" : "#/definitions/IndyAttrValue"
              } ],
              "description" : "Attribute value",
              "type" : "object"
            },
```
This is the only field in the swagger spec that has `additionalProperties` that looks like this. Usually it's just a ref, or just a type and description.

I removed the description metadata for the Nested IndyAttrValueSchema field, and now `additionalProperties` no longer has the type + description in Swagger 2.0, and now it converts correctly in OpenAPI 3. I'll create a PR shortly
